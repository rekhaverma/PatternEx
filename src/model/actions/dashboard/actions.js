export default {
  REFRESH_MALICIOUS_BEHAVIOR: '@@rest/REFRESH_MALICIOUS_BEHAVIOR',
  REFRESH_MALICIOUS_BEHAVIOR_COMPLETE: '@@rest/REFRESH_MALICIOUS_BEHAVIOR_COMPLETE',
  REFRESH_MALICIOUS_BEHAVIOR_FAILED: '@@rest/REFRESH_MALICIOUS_BEHAVIOR_FAILED',

  REFRESH_SUSPICIOUS_BEHAVIOR: '@@rest/REFRESH_SUSPICIOUS_BEHAVIOR',
  REFRESH_SUSPICIOUS_BEHAVIOR_COMPLETE: '@@rest/REFRESH_SUSPICIOUS_BEHAVIOR_COMPLETE',
  REFRESH_SUSPICIOUS_BEHAVIOR_FAILED: '@@rest/REFRESH_SUSPICIOUS_BEHAVIOR_FAILED',

  FETCH_SUMMARY: '@@rest/FETCH_SUMMARY',
  FETCH_SUMMARY_BY_CHUNK: '@@rest/FETCH_SUMMARY_BY_CHUNK',
  FETCH_SUMMARY_COMPLETE: '@@rest/FETCH_SUMMARY_COMPLETE',
  FETCH_SUMMARY_CHUNK_COMPLETE: '@@rest/FETCH_SUMMARY_CHUNK_COMPLETE',
  FETCH_SUMMARY_FAILED: '@@rest/FETCH_SUMMARY_FAILED',
  REFRESH_SUMMARY: '@@rest/REFRESH_SUMMARY',
  REFRESH_SUMMARY_COMPLETE: '@@rest/REFRESH_SUMMARY_COMPLETE',
  REFRESH_SUMMARY_CHUNK_COMPLETE: '@@rest/REFRESH_SUMMARY_CHUNK_COMPLETE',
  REFRESH_SUMMARY_FAILED: '@@rest/REFRESH_SUMMARY_FAILED',

  REFRESH_DASHBOARD: 'REFRESH_DASHBOARD',
};
