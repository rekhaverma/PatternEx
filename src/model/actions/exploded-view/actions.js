export default {
  FETCH_SEARCH_DATA: '@@explodedView/FETCH_SEARCH_DATA',
  SET_SEARCH_DATA: '@@explodedView/SET_SEARCH_DATA',
  SET_MODEL_NAME: '@@explodedView/SET_MODEL_NAME',

  FETCH_CLUSTER_RELATIONS: '@@explodedView/FETCH_CLUSTER_RELATIONS',
  SET_CLUSTER_RELATIONS: '@@explodedView/SET_CLUSTER_RELATIONS',

  SET_HISTORICAL_DATA: '@@explodedView/SET_HISTORICAL_DATA',

  SET_LABELS_HISTORY: '@@explodedView/SET_LABELS_HISTORY',
  FETCH_LABELS_HISTORY: '@@explodedView/FETCH_LABELS_HISTORY',

  FETCH_RELATED_ENTITIES_DATA: '@@explodedView/FETCH_RELATED_ENTITIES_DATA',
  SET_RELATED_ENTITIES_DATA: '@@explodedView/SET_RELATED_ENTITIES_DATA',

  GET_ENTITY_INFO: '@@explodedView/GET_ENTITY_INFO',

  GET_NX_DOMAINS: '@@explodedView/GET_NX_DOMAINS',

  FETCH_VULNERABILITY_REPORT_DATA: '@@explodedView/FETCH_VULNERABILITY_REPORT_DATA',
  SET_VULNERABILITY_REPORT_DATA: '@@explodedView/SET_VULNERABILITY_REPORT_DATA',

  SET_FW_LOGS_DATA: '@@explodedView/SET_FW_LOGS_DATA',
  SET_DNS_LOGS_DATA: '@@explodedView/SET_DNS_LOGS_DATA',

  GET_EDR_LOGS_DATA: '@@rest/GET_EDR_LOGS_DATA',
  GET_EDR_LOGS_DATA_SUCCESS: '@@rest/GET_EDR_LOGS_DATA_SUCCESS',
  GET_EDR_LOGS_DATA_FAILED: '@@rest/GET_EDR_LOGS_DATA_FAILED',

  SET_LABEL_START: '@@rest/SET_LABEL_START',
  SET_LABEL_COMPLETE: '@@rest/SET_LABEL_COMPLETE',

  GET_DHCP_DATA: '@@rest/GET_DHCP_DATA',
  GET_DHCP_DATA_SUCCESS: '@@rest/GET_DHCP_DATA_SUCCESS',
  GET_DHCP_DATA_FAILED: '@@rest/GET_DHCP_DATA_FAILED',

  GET_DETAILS_DATA: '@@rest/GET_DETAILS_DATA',
  SET_DETAILS_DATA: '@@rest/SET_DETAILS_DATA',
  GET_DETAILS_DATA_FAILED: '@@rest/GET_DETAILS_DATA_FAILED',
};
