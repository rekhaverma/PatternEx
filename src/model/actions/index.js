export * from './filters.actions';
export * from './rest.actions';
export * from './cluster.actions';
export * from './predictions.actions';
export * from './timeline.actions';
export * from './ui.actions';
export * from './rest/summary.actions';
export * from './rest/systemInfo.actions';
export * from './rest/pipelines.actions';
export * from './rest/performance.actions';
export * from './rest/models.actions';
export * from './rest/rules.actions';
export * from './rest/reports/reports.actions';
export * from './zeppelin.actions';
export * from './relations.actions';
export * from './ambari.actions';
export * from './rest/resource.actions.js';
export * from './rest/users.actions.js';
export * from './exploded-view.actions';
export * from './rest/upload-file.actions';
export * from './session.actions';
