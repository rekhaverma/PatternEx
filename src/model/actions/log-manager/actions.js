export default {
  GET_SUMMARY_DATA_BEGIN_LOADING: '@@logManager/GET_SUMMARY_DATA_BEGIN_LOADING',
  GET_SUMMARY_DATA_COMPLETE_LOADING: '@@logManager/GET_SUMMARY_DATA_COMPLETE_LOADING',

  GET_DATA_SOURCES_BEGIN_LOADING: '@@logManager/GET_DATA_SOURCES_BEGIN_LOADING',
  GET_DATA_SOURCES_COMPLETE_LOADING: '@@logManager/GET_DATA_SOURCES_COMPLETE_LOADING',
  GET_DATA_SOURCES_FAIL_LOADING: '@@logManager/GET_DATA_SOURCES_FAIL_LOADING',

  GET_DATA_CONFIG_BEGIN_LOADING: '@@logManager/GET_DATA_CONFIG_BEGIN_LOADING',
  GET_DATA_CONFIG_COMPLETE_LOADING: '@@logManager/GET_DATA_CONFIG_COMPLETE_LOADING',
  GET_DATA_CONFIG_FAIL_LOADING: '@@logManager/GET_DATA_CONFIG_FAIL_LOADING',

  ADD_DATA_SOURCE_BEGIN_SENDING: '@@logManager/ADD_DATA_SOURCE_BEGIN_SENDING',
  ADD_DATA_SOURCE_COMPLETE_SENDING: '@@logManager/ADD_DATA_SOURCE_COMPLETE_SENDING',
  ADD_DATA_SOURCE_FAIL_SENDING: '@@logManager/ADD_DATA_SOURCE_FAIL_SENDING',

  UPDATE_DATA_SOURCE_BEGIN_SENDING: '@@logManager/UPDATE_DATA_SOURCE_BEGIN_SENDING',
  UPDATE_DATA_SOURCE_COMPLETE_SENDING: '@@logManager/UPDATE_DATA_SOURCE_COMPLETE_SENDING',
  UPDATE_DATA_SOURCE_FAIL_SENDING: '@@logManager/UPDATE_DATA_SOURCE_FAIL_SENDING',

  START_DATA_SOURCE_BEGIN_SENDING: '@@logManager/START_DATA_SOURCE_BEGIN_SENDING',
  START_DATA_SOURCE_COMPLETE_SENDING: '@@logManager/START_DATA_SOURCE_COMPLETE_SENDING',
  START_DATA_SOURCE_FAIL_SENDING: '@@logManager/START_DATA_SOURCE_FAIL_SENDING',

  STOP_DATA_SOURCE_BEGIN_SENDING: '@@logManager/STOP_DATA_SOURCE_BEGIN_SENDING',
  STOP_DATA_SOURCE_COMPLETE_SENDING: '@@logManager/STOP_DATA_SOURCE_COMPLETE_SENDING',
  STOP_DATA_SOURCE_FAIL_SENDING: '@@logManager/STOP_DATA_SOURCE_FAIL_SENDING',

  DELETE_DATA_SOURCE_BEGIN_SENDING: '@@logManager/DELETE_DATA_SOURCE_BEGIN_SENDING',
  DELETE_DATA_SOURCE_COMPLETE_SENDING: '@@logManager/DELETE_DATA_SOURCE_COMPLETE_SENDING',
  DELETE_DATA_SOURCE_FAIL_SENDING: '@@logManager/DELETE_DATA_SOURCE_FAIL_SENDING',
};
