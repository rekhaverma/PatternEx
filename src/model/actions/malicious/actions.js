export default {
  FETCH_MALICIOUS_BEHAVIOR: '@@rest/FETCH_MALICIOUS_BEHAVIOR',
  FETCH_MALICIOUS_BEHAVIOR_COMPLETE: '@@rest/FETCH_MALICIOUS_BEHAVIOR_COMPLETE',
  FETCH_MALICIOUS_BEHAVIOR_FAILED: '@@rest/FETCH_MALICIOUS_BEHAVIOR_FAILED',
};
