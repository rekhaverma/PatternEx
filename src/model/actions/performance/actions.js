export default {
  FETCH_PERFORMANCE_DATA: '@@rest/FETCH_PERFORMANCE_DATA',
  FETCH_PERFORMANCE_DATA_COMPLETE: '@@rest/FETCH_PERFORMANCE_DATA_COMPLETE',
  FETCH_PERFORMANCE_DATA_FAILED: '@@rest/FETCH_PERFORMANCE_DATA_FAILED',
};
