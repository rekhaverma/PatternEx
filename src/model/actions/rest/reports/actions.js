export default {
  FETCH_REPORTS: '@@rest/FETCH_REPORTS',
  FETCH_REPORTS_SUCCESS: '@@rest/FETCH_REPORTS_SUCCESS',
  FETCH_REPORTS_FAILED: '@@rest/FETCH_REPORTS_FAILED',

  FETCH_REPORT_SUMMARY: '@@rest/FETCH_REPORT_SUMMARY',
  FETCH_REPORT_SUMMARY_SUCCESS: '@@rest/FETCH_REPORT_SUMMARY_SUCCESS',
  FETCH_REPORT_SUMMARY_FAILED: '@@rest/FETCH_REPORT_SUMMARY_FAILED',

  FETCH_REPORT_SUMMARY_DATES: '@@rest/FETCH_REPORT_SUMMARY_DATES',
  FETCH_REPORT_SUMMARY_DATES_SUCCESS: '@@rest/FETCH_REPORT_SUMMARY_DATES_SUCCESS',
  FETCH_REPORT_SUMMARY_DATES_FAILED: '@@rest/FETCH_REPORT_SUMMARY_DATES_FAILED',

  CREATE_NEW_REPORT: '@@rest/CREATE_NEW_REPORT',
  CREATE_NEW_REPORT_SUCCESS: '@@rest/CREATE_NEW_REPORT_SUCCESS',
  CREATE_NEW_REPORT_FAILED: '@@rest/CREATE_NEW_REPORT_FAILED',

  EDIT_REPORT: '@@rest/EDIT_REPORT',
  EDIT_REPORT_SUCCESS: '@@rest/EDIT_REPORT_SUCCESS',
  EDIT_REPORT_FAILED: '@@rest/EDIT_REPORT_FAILED',

  DELETE_REPORT: '@@rest/DELETE_REPORT',
  DELETE_REPORT_SUCCESS: '@@rest/DELETE_REPORT_SUCCESS',
  DELETE_REPORT_FAILED: '@@rest/DELETE_REPORT_FAILED',

  GET_REPORT_BYID: '@@rest/GET_REPORT_BYID',
  GET_REPORT_BYID_SUCCESS: '@@rest/GET_REPORT_BYID_SUCCESS',
  GET_REPORT_BYID_FAILED: '@@rest/GET_REPORT_BYID_FAILED',

  FETCH_REPORT_RESULTS: '@@rest/FETCH_REPORT_RESULTS',
  FETCH_REPORT_RESULTS_SUCCESS: '@@rest/FETCH_REPORT_RESULTS_SUCCESS',
  FETCH_REPORT_RESULTS_FAILED: '@@rest/FETCH_REPORT_RESULTS_FAILED',

  FETCH_REPORT_RESULTS_CSV: '@@rest/FETCH_REPORT_RESULTS_CSV',
  FETCH_REPORT_RESULTS_CSV_SUCCESS: '@@rest/FETCH_REPORT_RESULTS_CSV_SUCCESS',
  FETCH_REPORT_RESULTS_CSV_FAILED: '@@rest/FETCH_REPORT_RESULTS_CSV_FAILED',

  RESET_REPORTS_DETAIL_STATE: '@@setter/RESET_REPORTS_DETAIL_STATE',
  RESET_DOWNLOAD_CSV: '@@setter/RESET_DOWNLOAD_CSV',

  ON_DEMAND: '@@rest/ON_DEMAND',
  ON_DEMAND_SUCCESS: '@@rest/ON_DEMAND_SUCCESS',
  ON_DEMAND_FAILED: '@@rest/ON_DEMAND_FAILED',
};
