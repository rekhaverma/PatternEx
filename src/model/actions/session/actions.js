export default {
  USER_IS_LOGGED_IN_START: '@@session/USER_IS_LOGGED_IN_START',
  USER_IS_LOGGED_IN_SUCCESS: '@@session/USER_IS_LOGGED_IN_SUCCESS',
  USER_IS_LOGGED_IN_FAIL: '@@session/USER_IS_LOGGED_IN_FAIL',

  USER_LOGIN_START: '@@session/USER_LOGIN_START',
  USER_LOGIN_SUCCESS: '@@session/USER_LOGIN_SUCCESS',
  USER_LOGIN_FAIL: '@@session/USER_LOGIN_FAIL',

  USER_LOGOUT_START: '@@session/USER_LOGOUT_START',
  USER_LOGOUT_SUCCESS: '@@session/USER_LOGOUT_SUCCESS',
  USER_LOGOUT_FAIL: '@@session/USER_LOGOUT_FAIL',
};
