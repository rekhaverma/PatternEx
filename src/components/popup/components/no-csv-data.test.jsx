import React from 'react';
import { NoCSVData, mapDispatchToProps } from './nocsvdata.component';
import { SET_POPUP } from 'model/actions/ui.actions';

/**
 * @status: Complete
 * @sign-off-by: Alex Andries
 */
describe('<NoCSVData />', () => {
  describe('renders: ', () => {
    let wrapper; let props;

    beforeEach(() => {
      props = {
        className: 'no-csv-data-popup',
        text: <div className="no-csv-data-popup__text">Text</div>,
        onClick: jest.fn(),
      };
      wrapper = shallow(<NoCSVData {...props} />);
    })

    it('should match with snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should call onClick when element is clicked', () => {
      wrapper.find('.button').simulate('click');
      expect(props.onClick.mock.calls.length).toBe(1);
    });
  });

  describe('mapDispatchToProps method', () => {
    it('should trigger the setPopup action', () => {
      const dispatch = jest.fn();
      mapDispatchToProps(dispatch).onClick('');

      expect(dispatch.mock.calls[0][0]).toEqual({
        type: SET_POPUP,
        payload: '',
      });
    });
  });
}) 
