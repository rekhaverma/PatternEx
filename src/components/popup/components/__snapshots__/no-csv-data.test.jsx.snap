// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NoCSVData /> renders:  should match with snapshot 1`] = `
<div
  className="no-csv-data-popup"
>
  <div
    className="no-csv-data-popup__content"
  >
    <div
      className="no-csv-data-popup__text"
    >
      Text
    </div>
    <Button
      className="button"
      onClick={[Function]}
      style={
        Object {
          "width": 100,
        }
      }
    >
      Ok
    </Button>
  </div>
</div>
`;
